src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,215 @@
+            @Nonnull final SphereClientConfig clientConfig) {
+        return new RetryableSphereClientWithExponentialBackoff(clientConfig);
+    }
+
+    /**
+     * Sets the timeout value.
+     * @param timeout - build with timeout value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given timeout value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withTimeout(final long timeout) {
+        this.timeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the initialDelay value.
+     * @param initialDelay - build with initialDelay value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given initialDelay value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withInitialDelay(final long initialDelay) {
+        if (initialDelay < timeout) {
+            this.initialDelay = initialDelay;
+        }
+        return this;
+    }
+
+    /**
+     * Sets the timeUnit.
+     * @param timeUnit - build with timeUnit value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given timeUnit value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withTimeUnit(@Nonnull final TimeUnit timeUnit) {
+        this.timeUnit = timeUnit;
+        return this;
+    }
+
+    /**
+     * Sets the Max Retry value.
+     * @param maxRetries - build with maxRetries value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given maxRetries value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withMaxRetries(final int maxRetries) {
+        if (maxRetries > 1) {
+            this.maxRetries = maxRetries;
+        }
+        return this;
+    }
+
+    /**
+     * Sets the Max Parallel Requests value.
+     * @param maxParallelRequests - build with maxParallelRequests value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given maxParallelRequests value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withMaxParallelRequests(final int maxParallelRequests) {
+        this.maxParallelRequests = maxParallelRequests;
+        return this;
+    }
+
+    /**
+     * Sets the Retry Error Status Codes.
+     * @param retryErrorStatusCodes - build with retryErrorStatusCodes.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given retryErrorStatusCodes.
+     */
+    public RetryableSphereClientWithExponentialBackoff withRetryErrorStatusCodes(
+            final int[] retryErrorStatusCodes) {
+        this.retryErrorStatusCodes = retryErrorStatusCodes.clone();
+        return this;
+    }
+
+    /**
+     * creates a SphereClient using the class configuration values.
+     * @return {@link SphereClient}
+     */
+    public SphereClient build() {
+        return createClient();
+    }
+
+    /**
+     * Creates a {@link SphereClient}.
+     * @return the instantiated {@link SphereClient}.
+     */
+    protected SphereClient createClient() {
```

Comment:
do we need createClient methods or build is enough ?

Reply:
Build is enough, Removed this method. https://github.com/commercetools/commercetools-sync-java/pull/623/commits/ac84c9ecae58aba5365263b3dabf53cde7d9e971